src/Simulator/IncrementalRedstoneSimulator/RedstoneRepeaterHandler.h

```diff
@@ -71,4 +78,84 @@ class cRedstoneRepeaterHandler:
 	{
 		return { cBlockRedstoneRepeaterHandler::GetRearCoordinateOffset(a_Meta) + a_Position };
 	}
+
+private:
+
+	inline static bool IsOn(BLOCKTYPE a_Block)
+	{
+		return (a_Block == E_BLOCK_REDSTONE_REPEATER_ON);
+	}
+
+	/** Returns a pair with first element indicating if the block at the given position is an activated repeater.
+	If it is activated, the second element is the repeater metadata. */
+	static std::pair<bool, NIBBLETYPE> IsOnRepeater(cWorld & a_World, const Vector3i a_Position)
+	{
+		BLOCKTYPE Type;
+		NIBBLETYPE Meta;
+
+		if (!a_World.GetBlockTypeMeta(a_Position, Type, Meta))
+		{
+			return std::make_pair(false, 0);
+		}
+
+		return std::make_pair(IsOn(Type), Meta);
+	}
+
+	/** Determine if a repeater is locked.
+	A locked repeater is one with another powered repeater facing them, to their immediate left or right sides.
+	"Left" is relative to the direction the repeater output faces, naturally. */
+	inline static bool IsLocked(cWorld & a_World, const Vector3i a_Position, const NIBBLETYPE a_Meta)
+	{
+		// The left hand side offset. Will be negated to get the rhs offset
+		const auto LhsOffset = cBlockRedstoneRepeaterHandler::GetLeftCoordinateOffset(a_Meta);
+
+		// Test the block to the left of us
+		const auto Lhs = IsOnRepeater(a_World, LhsOffset + a_Position);
+		if (Lhs.first && DoesLhsLockMe(Lhs.second, a_Meta))
+		{
+			return true;
+		}
+
+		// Test the right side, flipping the argument order to DoesLhsLockMe
+		const auto Rhs = IsOnRepeater(a_World, -LhsOffset + a_Position);
+		return Rhs.first && DoesLhsLockMe(a_Meta, Rhs.second);
+	}
+
+	/** Determine, from the metadata of a repeater on our left side, if they lock us.
+	To test a repeater on our right, simply invert the order of arguments provided.
+	"Left" is relative to the direction the repeater output faces, naturally. */
+	static bool DoesLhsLockMe(NIBBLETYPE a_MetaLhs, NIBBLETYPE a_MyMeta)
+	{
+		// Get the direction bits
+		a_MetaLhs &= 0x03;
+		a_MyMeta &= 0x03;
+
+		// Convenience function to compare metadata
+		const auto CheckMetas = [&](NIBBLETYPE a_ExpectedLhs, NIBBLETYPE a_MyExpected)
```

Comment:
This function doesn't really help with the readability IMO, I would just inline it manually.

Reply:
Especially if you give the meta values symbolic names